ibis/common/egraph.py

```diff
@@ -94,6 +95,19 @@ def __getitem__(self, id) -> set[K]:
         id = self._parents[id]
         return self._classes[id]
 
+    def __delitem__(self, id):
+        """Remove the given id from the disjoint set.
+
+        Parameters
+        ----------
+        id :
+            The id to remove from the disjoint set.
+
+        """
+        self[id].remove(id)
```

Comment:
This is slightly more complicated than this, need to update the parent pointers.